metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -139,6 +153,86 @@ public TableDto create(final QualifiedName name, final TableDto tableDto) {
         return dto;
     }
 
+    private ObjectNode createParentChildObjectNode(@Nullable final Set<ParentInfo> parentInfos) {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        final ObjectNode rootNode = objectMapper.createObjectNode();
+
+        // For now, we only populate parent information for a child table.
+        if (parentInfos != null && !parentInfos.isEmpty()) {
+            final ArrayNode parentArrayNode = objectMapper.createArrayNode();
+            for (ParentInfo parentInfo : parentInfos) {
+                final ObjectNode parentNode = objectMapper.createObjectNode();
+                parentNode.put("name", parentInfo.getName());
+                parentNode.put("relationType", parentInfo.getRelationType());
+                parentNode.put("uuid", parentInfo.getUuid());
+                parentArrayNode.add(parentNode);
+            }
+            rootNode.set(ParentChildRelMetadataConstants.PARENTINFOS, parentArrayNode);
+        }
+        return rootNode;
+    }
+
+    // Return the unSaveOperation if applicable
+    private Optional<Runnable> saveParentChildRelationship(final QualifiedName child, final TableDto tableDto) {
+        if (tableDto.getDefinitionMetadata() != null) {
+            final ObjectNode definitionMetadata = tableDto.getDefinitionMetadata();
+            if (definitionMetadata.has(ParentChildRelMetadataConstants.PARENTNAME)) {
+                // fetch parent name
+                final String parentFullName = definitionMetadata.path(ParentChildRelMetadataConstants.PARENTNAME)
+                    .asText();
+                String parentUUID;
+                String childUUID;
+
+                if (!definitionMetadata.has(ParentChildRelMetadataConstants.PARENTUUID)) {
+                    throw new RuntimeException("root_table_uuid is not specified for parent table=" + parentFullName);
+                }
+
+                if (!definitionMetadata.has(ParentChildRelMetadataConstants.CHILDUUID)) {
+                    throw new RuntimeException("child_table_uuid is not specified for child table=" + child);
+                }
+                parentUUID = definitionMetadata.path(ParentChildRelMetadataConstants.PARENTUUID).asText();
+                childUUID = definitionMetadata.path(ParentChildRelMetadataConstants.CHILDUUID).asText();
+
+                final String[] splits = parentFullName.split("\\.");
```

Comment:
What's the reason to switch between the qualified name string using '/' as delimiter vs this way? Unifying on a single pattern like using '/' would be more uniform and not require conversions.

Reply:
good point, our internal client actually will pass a . separated. Probably we can change the client level behavior.